### DISCUSSION TITLE
having problem pranking fundme function

### DISCUSSION THREAD

**Discussion by QsAdedipe** [Cyfrin/foundry-full-course-cu#2598] [General]
why is my test code below failing, I think the prank cheat code is not working
"
    //arrange

        uint160 numberOfFunders = 10;
        uint160 startingFunderIndex = 1;
        for (uint160 i = startingFunderIndex; i < numberOfFunders; i++) {
            //vm.prank new address
            //vm. deal fund new ddress
            hoax(address(i), 10 ether);
            fundMe.fund{value: 10 ether}();
        }

        uint256 startingOwnerBalance = fundMe.getOwner().balance;
        uint256 startingFundMeBalance = address(fundMe).balance;

        console.log(startingOwnerBalance);
        console.log(startingFundMeBalance);

        //act
        vm.startPrank(fundMe.getOwner());
        //vm.prank(fundMe.getOwner());
        fundMe.withdraw();
        //vm.stopPrank();

        //Assert
        assertEq(address(fundMe).balance, 0);
        assertEq(
            startingFundMeBalance + startingOwnerBalance,
            fundMe.getOwner().balance
        );
    }
"
This is the result I am getting
"root@DESKTOP-IM9H78N:~/foundry-fund-me-f23# forge test --match-test testingWithDrawWithMultipleFunder -vvvv
[⠃] Compiling...
No files changed, compilation skipped

Ran 1 test for test/FundMeTest.t.sol:FundMeTest
[FAIL. Reason: EvmError: Revert] testingWithDrawWithMultipleFunder() (gas: 20480)
Traces:
  [2758275] FundMeTest::setUp()
    ├─ [1390748] → new DeployFundMe@0x5615dEB798BB3E4dFa0139dFa1b3D433Cc23b72f
    │   └─ ← [Return] 6836 bytes of code
    ├─ [1325903] DeployFundMe::run()
    │   ├─ [885598] → new HelperConfig@0x104fBc016F4bb334D775a19E8A6510109AC63E00
    │   │   ├─ [0] VM::startBroadcast()
    │   │   │   └─ ← [Return] 
    │   │   ├─ [377461] → new MockV3Aggregator@0x90193C961A926261B756D1E5bb255e67ff9498A1
    │   │   │   └─ ← [Return] 1108 bytes of code
    │   │   ├─ [0] VM::stopBroadcast()
    │   │   │   └─ ← [Return] 
    │   │   └─ ← [Return] 2237 bytes of code
    │   ├─ [359] HelperConfig::activeNetworkConfig() [staticcall]
    │   │   └─ ← [Return] 0x0000000000000000000000000000000000000000
    │   ├─ [0] VM::startBroadcast()
    │   │   └─ ← [Return] 
    │   ├─ [373367] → new FundMe@0xA8452Ec99ce0C64f20701dB7dD3abDb607c00496
    │   │   └─ ← [Return] 1742 bytes of code
    │   ├─ [0] VM::stopBroadcast()
    │   │   └─ ← [Return] 
    │   └─ ← [Return] FundMe: [0xA8452Ec99ce0C64f20701dB7dD3abDb607c00496]
    ├─ [0] VM::deal(user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D], 20000000000000000000 [2e19])
    │   └─ ← [Return] 
    └─ ← [Stop] 

  [20480] FundMeTest::testingWithDrawWithMultipleFunder()
    ├─ [0] VM::deal(0x0000000000000000000000000000000000000001, 10000000000000000000 [1e19])
    │   └─ ← [Return] 
    ├─ [0] VM::prank(0x0000000000000000000000000000000000000001)
    │   └─ ← [Return] 
    ├─ [5192] FundMe::fund{value: 10000000000000000000}()
    │   ├─ [0] 0x0000000000000000000000000000000000000000::latestRoundData() [staticcall]
    │   │   └─ ← [Stop] 
    │   └─ ← [Revert] EvmError: Revert
    └─ ← [Revert] EvmError: Revert

Suite result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 362.60ms (22.60ms CPU time)

Ran 1 test suite in 2.17s (362.60ms CPU time): 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/FundMeTest.t.sol:FundMeTest
[FAIL. Reason: EvmError: Revert] testingWithDrawWithMultipleFunder() (gas: 20480)
"

**Comment by EngrPips** [comment]
Hello @QsAdedipe, why is the address of your `price feed` contract address(0) as shown below.

```js
├─ [359] HelperConfig::activeNetworkConfig() [staticcall]
│ │ └─ ← [Return] 0x0000000000000000000000000000000000000000
```

Please show your `HelperConfig`

**Comment by QsAdedipe** [comment]
This is my HelperConfig
"//SPDX-License-Identifier: SEE LICENSE IN LICENSE
//Deploy mocks when we are on a local anvil chain
//Keep track of contract address across different cchains
//Sepolia ETH/USD
//Mainnet ETH/USD
pragma solidity ^0.8.25;

import {Script} from "forge-std/Script.sol";
import {MockV3Aggregator} from "../test/mocks/MockV3Aggregator.sol";

contract HelperConfig is Script {
    //if we are on a local anvil, we deploy mocks
    //otherwise, grab the existing address from the live network

    NetworkConfig public activeNetworkConfig;
    uint8 public constant DECIMALS = 8;
    int256 public constant INITIAL_PRICE = 2000e8;

    struct NetworkConfig {
        address priceFeed; //ETH/USD price address
    }

    constructor() {
        if (block.chainid == 11155111) {
            activeNetworkConfig = getSepoliaEthConfig();
        } else if (block.chainid == 1) {
            activeNetworkConfig = getMainnetEthConfig();
        } else {
            activeNetworkConfig = getOrCreateAnviEthConfig();
        }
    }

    function getSepoliaEthConfig() public pure returns (NetworkConfig memory)
{
        //price feed address
        NetworkConfig memory sepoliaConfig = NetworkConfig({
            priceFeed: 0x694AA1769357215DE4FAC081bf1f309aDC325306
        });

        return sepoliaConfig;
    }

    function getMainnetEthConfig() public pure returns (NetworkConfig memory)
{
        //price feed address
        NetworkConfig memory ethConfig = NetworkConfig({
            priceFeed: 0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419
        });

        return ethConfig;
    }

    //anonymousfunction getAnviEthConfig() public pure returns
(NetworkConfig memory) {}

    function getOrCreateAnviEthConfig() public returns (NetworkConfig memory)
{
        // price feed address

        //1. Deploy the mocks ("fake contract")
        //2. Return the mock address
        if (activeNetworkConfig.priceFeed != address(0)) {
            return activeNetworkConfig;
        }
        vm.startBroadcast();
        MockV3Aggregator mockPriceFeed = new MockV3Aggregator(
            DECIMALS,
            INITIAL_PRICE
        );

        vm.stopBroadcast();

        NetworkConfig memory AnvilConfig = NetworkConfig({
            priceFeed: address(mockPriceFeed)
        });
    }
}


On Wed, Oct 2, 2024 at 8:31 AM EngrPips ***@***.***> wrote:

> Hello @QsAdedipe <https://github.com/QsAdedipe>, why is the address of
> your price feed contract address(0) as shown below.
>
> ├─ [359] HelperConfig::activeNetworkConfig() [staticcall]│ │ └─ ← [Return] 0x0000000000000000000000000000000000000000
>
> Please show your HelperConfig
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Cyfrin/foundry-full-course-cu/discussions/2598#discussioncomment-10816958>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXKXH2OAPS7D5MNN6RYG6O3ZZOOLRAVCNFSM6AAAAABPGQ7NMKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAOBRGY4TKOA>
> .
> You are receiving this because you were mentioned.Message ID:
> <Cyfrin/foundry-full-course-cu/repo-discussions/2598/comments/10816958@
> github.com>
>


-- 
QS Adedipe Oluwasina, MNIQS, RQS

**Reply by EngrPips** [reply]
Please change your `getOrCreateAnviEthConfig` to the below and let's see if it helps

```js
function getOrCreateAnviEthConfig() public returns (NetworkConfig memory AnvilConfig)
{
        // price feed address

        //1. Deploy the mocks ("fake contract")
        //2. Return the mock address
        if (activeNetworkConfig.priceFeed != address(0)) {
            return activeNetworkConfig;
        }
        vm.startBroadcast();
        MockV3Aggregator mockPriceFeed = new MockV3Aggregator(
            DECIMALS,
            INITIAL_PRICE
        );

        vm.stopBroadcast();

         AnvilConfig = NetworkConfig({
            priceFeed: address(mockPriceFeed)
        });
    }
```

**Comment by QsAdedipe** [comment]
Fixed!
Thank you very much, but I will like to ask, why did this happen?
because the previous code line followed Patric's instructions from the
course
is there a kind of update or what, I will like to know

On Wed, Oct 2, 2024 at 12:30 PM EngrPips ***@***.***> wrote:

> Please change your getOrCreateAnviEthConfig to the below and let's see if
> it helps
>
> function getOrCreateAnviEthConfig() public returns (NetworkConfig memory AnvilConfig){
>         // price feed address
>
>         //1. Deploy the mocks ("fake contract")
>         //2. Return the mock address
>         if (activeNetworkConfig.priceFeed != address(0)) {
>             return activeNetworkConfig;
>         }
>         vm.startBroadcast();
>         MockV3Aggregator mockPriceFeed = new MockV3Aggregator(
>             DECIMALS,
>             INITIAL_PRICE
>         );
>
>         vm.stopBroadcast();
>
>          AnvilConfig = NetworkConfig({
>             priceFeed: address(mockPriceFeed)
>         });
>     }
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Cyfrin/foundry-full-course-cu/discussions/2598#discussioncomment-10819484>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXKXH2LU4ZJ4VXXYQVDNCKTZZPKPFAVCNFSM6AAAAABPGQ7NMKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAOBRHE2DQNA>
> .
> You are receiving this because you were mentioned.Message ID:
> <Cyfrin/foundry-full-course-cu/repo-discussions/2598/comments/10819484@
> github.com>
>


-- 
QS Adedipe Oluwasina, MNIQS, RQS

**Reply by EngrPips** [reply]
I think you missed some details. Maybe rewatch the lesson and go slowly, and you might be able to catch the details you missed. I didn't know of any update.

**Comment by QsAdedipe** [comment]
Thanks, I'll check it out. To

**Reply by EngrPips** [reply]
Yeah do that.
